modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -402,6 +417,28 @@ protected Experiment internalGetExperiment(Application.Name appName,
     }
 
     /**
+     * Updates the tags for a given Application.
+     *
+     * @param applicationName the Application for which the tags should be added
+     * @param tags            list of tags for the experiment
+     */
+    protected void updateExperimentTags(Application.Name applicationName, Set<String> tags) {
+
+        if (tags == null)
+            tags = Collections.EMPTY_SET;
+
+        // update the tag table, just rewrite the complete entry
+        List<ExperimentTagsByApplication> listAllTags = experimentAccessor.getAllTags(applicationName.toString()).all();
```

Comment:
@abhishekjain88 : no worries, thanks for your input! I am not sure if I understood your concern though. Let me explain my reasoning. The reason for this code was my following thought (correct me if I am wrong): imagine you have several experiments for one application. Two of them have the following tags exp1 - [a,b], exp2 - [a,b,c]. The index table that holds the mapping between applications and their tags is now holding the tags that are present for that application app1 : [a,b,c,d,f,....]. What happens in the case that exp2 has a label removed e.g. b? After the experiment was updated, the index table has to be updated too. Now it would be not too smart to just remove the tag 'b' from the index table, since exp1 is still holding that tag. Instead of having now a code that checks for all combinations of possible updates and strange edge-cases, I decided to let the experiment table be the ground truth for the values of the index. You wrote that this should ideally not happen, but this method is exactly there for filling the `experiment_tag` table. Does that make sense?